drivers/lhal/src/bflb_usb_v1.c

```diff
@@ -266,6 +315,10 @@ int usb_dc_init(void)
 
     regval = 0;
     regval |= USB_CR_USB_RESET_EN;
+    /* CMD_* interrupts added to enabled set. They are needed in advanced bflb_usb_ep_set_ready() logic */
+    regval |= USB_CR_EP0_SETUP_CMD_EN;
```

Comment:
disable these irqs because too many irq enter and leave.

Reply:
BL702 USB device controller EP0 hardware interface requires this interrupt to allow proper USB control transaction processing, especially control **IN**. In this case host sends USB **SETUP** request to device (handled properly), BL702 device gets **IN** token and sends requested data (handled properly) and after this hosts send short **OUT** status packet to the device. Sometimes it is followed by next **SETUP** if we use fast USB Host or it is intensive USB HID data exchange. BL702 has only one common FIFO with single data size register and due to this restrictions software driver can't properly handle two consecutive data transfers (**OUT** closely followed by **SETUP**) from USB Host to the BL702 USB device. So in this case we some times will get **SETUP** xfer data length while trying to process short **OUT** packet in status stage of first USB transaction, closely followed by next control transfer. It's hardware race condition. Enabling EP0 CMD interrupts makes possible to detect such condition and handle it properly. At the same time I think that it is not problem and could not make too many IRQ enter/leave, cause USB control transfer is supposed to be slow and uses relatively rare (f.e. USB HID, etc.). We've implemented HID driver over EP0 (control transfer) and it couldn't perform properly without provided SDK modifications.